lib/template-parts.php

```diff
@@ -217,49 +218,6 @@ function filter_rest_wp_template_part_query( $args, $request ) {
 			'value' => $request['theme'],
 		);
 
-		// Ensure auto-drafts of all theme supplied template parts are created.
-		if ( wp_get_theme()->stylesheet === $request['theme'] ) {
```

Comment:
I removed just this part here but there's also the "template" and "resolved" query handling which I'm unsure about. What is it for and can we get rid of it? cc @ockham

Reply:
I actually went ahead and removed the "template" and "resolved" query handling cause it seems useless and used in Gutenberg? am I wrong?

Reply:
> I removed just this part here but there's also the "template" and "resolved" query handling which I'm unsure about. What is it for and can we get rid of it? cc @ockham

Both are fairly crucial for the functioning the Site Editor, and it's likely to break without them (unless the implementation has changed since I've moved to a different team -- @vindl @Addison-Stavlo et al will know more).

They basically add a way to indicate to the template and template part CPT entities to only return the object that's in charge of rendering a specific page (or, more generally, any kind of post type/taxonomy/archive/...). The `findTemplate` client side function uses the `resolved` arg here: https://github.com/WordPress/gutenberg/blob/90a70fd005f571d81abe66d00f18825a2b309a6b/packages/edit-site/src/utils/find-template.js#L30

Reply:
As you can see here, that particular instance is for `findTemplate` and not `findTemplatePart` so the flag support for template parts is not needed IMO. 

I also think that flag might not be that necessary for templates (with a bunch of frontend refactoring) but I'm leaving that separate.